### PR TITLE
Revert explain attributes change

### PR DESCRIPTION
**Pull request by adarshsanjeev**
Reverts a change made to explain attributes made by https://github.com/apache/druid/pull/15689. This will maintain backward compatibility. 
The `targetDataSource` attribute is changed to return a string containing the datasources name, or "extern" if the query is an export.

<hr>

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: sql/src/main/java/org/apache/druid/sql/calcite/planner/IngestHandler.java [hunk 1]
```diff
@@ -405,7 +405,7 @@ public ExplainAttributes explainAttributes()
     {
       return new ExplainAttributes(
           DruidSqlReplace.OPERATOR.getName(),
-          targetDatasource,
+          targetDatasource.getType(),
```
Comment (cryptoe) [inline sql/src/main/java/org/apache/druid/sql/calcite/planner/IngestHandler.java:408]: I find it weird that the type is returning the name of the table in tableDestination. Wdyt ? Should we adjust the inteface `IngestDestination` ?

Comment (gianm) [reply sql/src/main/java/org/apache/druid/sql/calcite/planner/IngestHandler.java:408]: This does seem like a mistake.

**Comment by gianm** [comment]
We did release #15689 in 29.0.0. It accidentally included a breaking API change, which is problematic. This PR reverts the breaking change, but since #15689 has already been released, that means this PR is _also_ a breaking API change. We need to treat it as such. There's a few options:

1. Don't make this change, just accept that #15689 had an unintentional breaking change, and keep rolling with the new format.
2. Make this change, and document in the release notes that 29.0.0 had a different format for `targetDataSource` from both prior and future versions.
3. Provide a migration path: add a query context property that controls the format of `targetDataSource` and allow people to override it.

IMO, given that this API is not likely widely used, and 29.0.0 was just a single release, (2) is the best option. But we do need to consider the other options when reviewing.

**Review by cryptoe — APPROVED** [review]
Changes LGTM. 
I agree with @gianm proposal. Lets go ahead with point #2.

**Pull request merged**